sips/sip-010/sip-010-fungible-token-standard.md

```diff
@@ -0,0 +1,176 @@
+| ---------- | ----------------------------------------------- |
+| u1         | `sender` does not have enough balance           |
+| u2         | `sender` and `recipient` are the same principal |
+| u3         | `amount` is non-positive                        |
+| u4         | `sender` is not the same as `tx-sender`         |
+
+### Name
+
+`(name () (response (string-ascii 32) uint))`
+
+Return a human-readable name for the contract, such as "CoolPoints", etc.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Symbol
+
+`(symbol () (response (string-ascii 32) uint))`
+
+Return a symbol that allows for a shorter representation of a token. This is sometimes referred to as a "ticker". Examples: "STX", "COOL", etc. Typically, a token could be referred to as $SYMBOL when referencing it in writing.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Decimals
+
+`(decimals () (response uint uint))`
+
+The number of decimal places in a token. All fungible token balances must be represented as integers, but providing the number of decimals provides for an abstraction of a token that humans are more familiar dealing with. For example, the US Dollar has 2 decimals, if the base unit is "cents", as is typically done in accounting. Stacks has 6 decimals, Bitcoin has 8 decimals, and so on.
+
+As another example, if a token has 4 decimals, and the `balance-of` a particular user returns `100345000`, wallets and exchanges would likely represent that value as `10034.5`.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Balance of
+
+`(balance-of (principal) (response uint uint))`
+
+Return the balance of a particular principal (also known as "address" or "account"). Implementations should typically use the built-in Clarity method `ft-get-balance`.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+### Total supply
+
+`(total-supply () (response uint uint))`
+
+Return the total supply of this token. Implementations should typically use the built-in Clarity method `ft-get-supply`.
+
+This method should be defined as read-only, i.e. `define-read-only`.
+
+## Trait implementation
+
+An implementation of the proposed trait is provided below.
+
+```clarity
+(define-trait ft-trait
+  (
+    ;; Transfer from the caller to a new principal
+    (transfer (uint principal principal) (response bool uint))
+
+    ;; the human readable name of the token
+    (name () (response (string-ascii 32) uint))
+
+    ;; the ticker symbol, or empty if none
+    (symbol () (response (string-ascii 32) uint))
+
+    ;; the number of decimals used, e.g. 6 would mean 1_000_000 represents 1 token
+    (decimals () (response uint uint))
+
+    ;; the balance of the passed principal
+    (balance-of (principal) (response uint uint))
+
+    ;; the current total supply (which does not need to be a constant)
+    (total-supply () (response uint uint))
+  )
+)
```

Comment:
Because trait methods _must_ return a `(response ..)` type, it might be worth mentioning here that some of these methods (like `symbol` and `decimals`) are expected to be "fail-proof" methods, and that an `(err ..)` return should be treated as a panic-worthy condition.  Up to you though.